### PR TITLE
nixos/systemd: add {pre,post}StartElevated

### PR DESCRIPTION
**Pull request by mkg20001**
###### Motivation for this change

Convenience option for the lazy

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by aanderse** [comment]
Please provide a **motivating** motivation section. See prior discussion [here](https://discourse.nixos.org/t/systemd-services-should-allow-running-commands-as-root/5036).

**Comment by flokli** [comment]
The docs should also mention this comes before other ExecStartPre options.

**Comment by aanderse** [comment]
For convenience I'll copy/paste what I said in the linked discussion, on why I **don't** think these options are a good idea:

> Just an opinion but… I think we should continue to not offer a convenient way to run commands as root for systemd services which don’t run as root. I don’t like the idea of promoting running preStart scripts as root when there are better ways to achieve the same results. In the few cases where there are no better alternatives having to use a special prefix seems like an acceptable cost.

**Comment by mkg20001** [comment]
https://github.com/NixOS/nixpkgs/pull/144174#discussion_r741922906

tl;dr

> When there was a convenient option (PermissionsStartOnly = true;) almost every single NixOS module used it, even when there was absolutely no need to. Forcing developers to think and learn the proper way to do things seems like a good idea to me in exchange for 1 in 100 modules needing an inconvenient use of ExecStartPre = "+" + (pkgs.writeScript "init.sh" "code to run as root")

**Comment by roberth** [comment]
This module needs to model the existing config file more accurately before adding conveniences.
`ExecStart` and such should be a list of submodules and those submodules should generate the `+` and whatnot based on understandable, mergeable etc configs.

**Comment by Infinisil** [comment]
I agree with @roberth, and I think this PR was merged way too soon, especially for a self-merge..

**Comment by aanderse** [comment]
@Infinisil it wasn't merged... closed as the author decided not to pursue it :wink:

**Comment by Infinisil** [comment]
Oh! This new GitHub color messed me up! Thanks @aanderse :)
